### PR TITLE
Fix case where multiple component ids break `--` (now reflects JS code)

### LINKED ISSUE #1247: (CaseClauseError) no case clause matching: nil in diff.ex
### Environment

* Elixir version (elixir -v):
Erlang/OTP 23 [erts-11.0.2] [source] [64-bit] [smp:24:24] [ds:24:24:10] [async-threads:1] [hipe]
                                                                                                
Elixir 1.11.0 (compiled with Erlang/OTP 21)                                                     

* Phoenix version (mix deps): locked at 1.5.7 (phoenix) 774cd644
* Phoenix LiveView version (mix deps): locked at 0.15.0 (phoenix_live_view) d42ad9b5
* NodeJS version (node -v): v14.15.0
* NPM version (npm -v): 6.14.8
* Operating system: Arch Linux (but also on CI, which is Ubuntu?)
* Browsers you attempted to reproduce this bug on (the more the merrier): Firefox (could not reproduce in-browser)
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no:

### Actual behavior
After upgrading to live_view 0.15.0, some of my live view tests are failing.
The relevant line in the tests: `view |> element("#set_rate_usps_priority") |> render_click()` (the `render_click` in this case)

Stacktrace:
```14:25:54.395 [error] GenServer #PID<0.12023.0> terminating                                                                                  
** (CaseClauseError) no case clause matching: nil                                                                                           
    (phoenix_live_view 0.15.0) lib/phoenix_live_view/diff.ex:84: Phoenix.LiveView.Diff.resolve_components_xrefs/2                           
    (phoenix_live_view 0.15.0) lib/phoenix_live_view/diff.ex:56: Phoenix.LiveView.Diff.to_iodata/3                                          
    (phoenix_live_view 0.15.0) lib/phoenix_live_view/diff.ex:70: Phoenix.LiveView.Diff.one_to_iodata/6                                      
    (phoenix_live_view 0.15.0) lib/phoenix_live_view/diff.ex:57: Phoenix.LiveView.Diff.to_iodata/3                                          
    (phoenix_live_view 0.15.0) lib/phoenix_live_view/diff.ex:70: Phoenix.LiveView.Diff.one_to_iodata/6                                      
    (phoenix_live_view 0.15.0) lib/phoenix_live_view/diff.ex:41: Phoenix.LiveView.Diff.to_iodata/2                                          
    (phoenix_live_view 0.15.0) lib/phoenix_live_view/test/dom.ex:189: Phoenix.LiveViewTest.DOM.render_diff/1                                
    (phoenix_live_view 0.15.0) lib/phoenix_live_view/test/client_proxy.ex:584: Phoenix.LiveViewTest.ClientProxy.merge_rendered/3            
    (phoenix_live_view 0.15.0) lib/phoenix_live_view/test/client_proxy.ex:666: Phoenix.LiveViewTest.ClientProxy.handle_reply/2              
    (phoenix_live_view 0.15.0) lib/phoenix_live_view/test/client_proxy.ex:345: Phoenix.LiveViewTest.ClientProxy.handle_info/2               
    (stdlib 3.13) gen_server.erl:680: :gen_server.try_dispatch/4                                                                            
    (stdlib 3.13) gen_server.erl:756: :gen_server.handle_msg/6                                                                              
    (stdlib 3.13) proc_lib.erl:226: :proc_lib.init_p_do_apply/3                                                                             
Last message: {:EXIT, #PID<0.12021.0>, {{:case_clause, nil}, [{Phoenix.LiveView.Diff, :resolve_components_xrefs, 2, [file: 'lib/phoenix_live
_view/diff.ex', line: 84]}, {Phoenix.LiveView.Diff, :to_iodata, 3, [file: 'lib/phoenix_live_view/diff.ex', line: 56]}, {Phoenix.LiveView.Dif
f, :one_to_iodata, 6, [file: 'lib/phoenix_live_view/diff.ex', line: 70]}, {Phoenix.LiveView.Diff, :to_iodata, 3, [file: 'lib/phoenix_live_vi
ew/diff.ex', line: 57]}, {Phoenix.LiveView.Diff, :one_to_iodata, 6, [file: 'lib/phoenix_live_view/diff.ex', line: 70]}, {Phoenix.LiveView.Di
ff, :to_iodata, 2, [file: 'lib/phoenix_live_view/diff.ex', line: 41]}, {Phoenix.LiveViewTest.DOM, :render_diff, 1, [file: 'lib/phoenix_live_
view/test/dom.ex', line: 189]}, {Phoenix.LiveViewTest.ClientProxy, :merge_rendered, 3, [file: 'lib/phoenix_live_view/test/client_proxy.ex', 
line: 584]}, {Phoenix.LiveViewTest.ClientProxy, :handle_reply, 2, [file: 'lib/phoenix_live_view/test/client_proxy.ex', line: 666]}, {Phoenix
.LiveViewTest.ClientProxy, :handle_info, 2, [file: 'lib/phoenix_live_view/test/client_proxy.ex', line: 345]}, {:gen_server, :try_dispatch, 4
, [file: 'gen_server.erl', line: 680]}, {:gen_server, :handle_msg, 6, [file: 'gen_server.erl', line: 756]}, {:proc_lib, :init_p_do_apply, 3,
 [file: 'proc_lib.erl', line: 226]}]}}                                                                                                      
```
I added some debug logging here:
 ```
defp resolve_components_xrefs(cid, components) do
+    IO.inspect({cid, components}, label: "CID")    
     case components[cid] do                        
```

This results in the following output right immediately prior to the error:
```
CID: {5,                                                                                                                                                                                                                                                                                  
 %{                                                                                                                                         
   1 => %{                                                                                                                                   
     0 => "collapse-slider-checkout",                                                                                                        
     1 => "",                                                                                                                                
     2 => "",                                                                                                                               
     3 => "checkout",                                                                                                                       
     4 => "checkout",                                                                                                                                                                                                                                                                     
     5 => "checkout",                                                                                                                       
     6 => "",                                                                                                                               
     7 => "d-flex",                                                                                                                         
     8 => "",                                                                                                                               
     9 => "vh-fill",                                                                                                                        
     10 => %{                                                                                                                               
       0 => "checkout d-flex flex-column vh-fill",                                                                                          
       1 => %{                                                                                                                              
         0 => %{                                                                                                                            
           0 => %{                                                                                                                          
             0 => 4,                                                                                                                        
             1 => "flex-grow-1 overflow-auto",                                                                                              
             2 => %{                                                  
               0 => %{d: [[2], [3]], s: ["\n      ", "\n    "]},                                                                            
               :s => ["\n    ", "\n  "]                                                                                                     
             },                                                                                                                             
             3 => %{                                                                                                                        
               0 => %{                                                                                                                      
                 0 => %{                                                                                                                    
                   0 => "<dl class=\"summary-price d-flex justify-content-between\"><dt>Subtotal</dt><dd>$20</dd></dl>",                    
                   1 => "",                                                                                                                 
                   2 => "<dl class=\"summary-price d-flex justify-content-between\"><dt>Shipping</dt><dd>&mdash;</dd></dl>",                
                   3 => "<dl class=\"summary-price d-flex justify-content-between\"><dt>Tax</dt><dd>$0</dd></dl>",                                                                                                                                                                        
                   4 => "",                                                                                                                                                                                                                                                               
                   5 => "<dl class=\"summary-price d-flex justify-content-between\"><dt>Total</dt><dd>$20</dd></dl>",                       
                   :s => ["\n      ", "\n\n      ", "\n\n      ", "\n      ",                                                               
                    "\n\n      ", "\n\n      ", "\n  "]                                                                                     
                 },                                                                                                                         
                 1 => %{                                                                                                                    
                   0 => %{                                                                                                                  
                     s: ["\n      <button type=\"button\" class=\"btn btn-primary w-100 btn-checkout\" phx-click=\"next_tab\">Checkout</button>\n    "]
                   },                                                                                                                       
                   :s => ["\n    ", "\n  "]                                                                                                 
                 },                                                                                                                         
                 :s => ["<div class=\"cart-summary\">\n  ", "\n\n  ",                                                                       
                  "\n</div>\n"]                                                                                                             
               },                                                                                                                           
               :s => ["\n  <div class=\"bottom-holder flex-shrink-1\">\n    ",                                                              
                "\n  </div>\n"]                                                                                                             
             },                                                                                                                             
             :s => ["\n          ", "\n        "]                                                                                           
           },                                                                                                                               
           1 => %{0 => 5, :s => ["\n          ", "\n        "]},                                                                            
           2 => %{0 => 6, :s => ["\n          ", "\n        "]},        
           3 => %{0 => 7, :s => ["\n          ", "\n        "]},
           4 => 8,                                                    
           :s => ["\n        ", "\n\n        ", "\n\n        ", "\n\n        ",                                                              
            "\n\n        ", "\n    "]
         },                                                           
         :s => ["\n    ", "\n  "]                                     
       },                                                             
       :s => ["\n      <div class=\"", "\">", "</div>\n    "]
     },                                                               
     :s => ["<div class=\"collapse-slider ", " collapse ", " ",
      "\" phx-hook=\"CollapseSlider\" data-phx-id=\"",
      "\" id=\"collapseSlider-",                                      
      "\">\n  <a href=\"#\" class=\"close-btn collapse-slider-opener\" phx-click=\"toggle_",                                                 
      "\"></a>\n\n  ", "\n\n  <div class=\"slide-body ", " ", " ", "\">\n    ",                                                              
      "\n  </div>\n</div>\n"]                                         
   },                                                                 
   4 => %{                                                            
     0 => %{                                                          
       0 => "checkout__heading checkout--success",
       1 => "phx-click=set_tab phx-value-tab=basic",
       2 => "Basic",                                                  
       3 => "",                                                       
       :s => ["<div class=\"", "\" ",
        ">\n  <strong class=\"slide-subtitle with-icon FTBase-medium-22-40\">",                                                              
        "</strong>\n</div>\n", "\n"]
     },                                                               
     :s => ["", "\n"]                                                 
   },                                                                 
   6 => %{                                                            
     0 => %{                                                          
       0 => "checkout__heading checkout--success",
       1 => "",                                                       
       2 => "Delivery",                                               
       3 => %{                                                        
         0 => %{                                                      
           0 => %{                                                    
             d: [                                                     
               [                                                      
                 "usps_priority",                                     
                 "usps_priority",                                     
                 "6",                                                 
                 "<input id=\"order_rate_usps_priority\" name=\"order[rate]\" type=\"radio\" value=\"usps_priority\" checked>",              
                 "Standard",                                          
                 %{                                                   
                   0 => "(5-7 Business Days)",
                   :s => ["\n                    ", "\n                  "]                                                                  
                 },                                                   
                 "$10"                                                
               ],                                                     
               [                                                      
                 "usps_priority_express",
                 "usps_priority_express",
                 "6",                                                 
                 "<input id=\"order_rate_usps_priority_express\" name=\"order[rate]\" type=\"radio\" value=\"usps_priority_express\">",      
                 "Express",                                           
                 %{                                                   
                   0 => "(3 Business Days)",
                   :s => ["\n                    ", "\n                  "]                                                                  
                 },                                                   
                 "$25"                                                
                     ]                                                      
             ],                                                       
             s: ["\n            <li>\n              <label class=\"default-radio\" phx-click=\"set_rate\" id=\"set_rate_",                   
              "\" phx-value-id=\"", "\" phx-target=\"", "\">\n                ",                                                             
              "\n                <span class=\"fake-input\"></span>\n                <span class=\"fake-label\">\n                  ",       
              "\n\n                  ",
              "\n\n                  <br />\n\n                  ",
              "\n                </span>\n              </label>\n            </li>\n          "]                                            
           },                                                         
           1 => "6",                                                  
           2 => "",                                                   
           :s => ["\n  <div class=\"checkout__slide flex-grow-1 overflow-auto\">\n    <div class=\"wrap\">\n      <div class=\"address-variation m-0\">\n        <ul class=\"list-unstyled check-list\">\n          ",
            "\n        </ul>\n      </div>\n    </div>\n  </div>\n  <div class=\"bottom-holder\">\n    <button class=\"btn btn-primary w-100 btn-checkout\" id=\"submit_delivery\" phx-click=\"submit\" phx-target=\"",
            "\" ", ">\n      Choose payment method\n    </button>\n  </div>\n"]                                                              
         },                                                           
         :s => ["\n  ", "\n"]                                         
       },                                                             
       :s => ["<div class=\"", "\" ",
        ">\n  <strong class=\"slide-subtitle with-icon FTBase-medium-22-40\">",                                                              
        "</strong>\n</div>\n", "\n"]
     },                                                               
     :s => ["", "\n"]                                                 
   },                                                                 
   7 => %{                                                            
     0 => %{                                                          
       0 => "checkout__heading",                                      
       1 => "",                                                       
       2 => "Payment",                                                
       3 => "",                                                       
       :s => ["<div class=\"", "\" ",
        ">\n  <strong class=\"slide-subtitle with-icon FTBase-medium-22-40\">",                                                              
        "</strong>\n</div>\n", "\n"]
     },                                                               
     :s => ["", "\n"]                                                 
   },                                                                 
   8 => %{                                                            
     0 => %{                                                          
       0 => "checkout__heading",                                      
       1 => "",                                                       
       2 => "Confirm",                                                
       3 => "",                                                       
       :s => ["<div class=\"", "\" ",
        ">\n  <strong class=\"slide-subtitle with-icon FTBase-medium-22-40\">",                                                              
        "</strong>\n</div>\n", "\n"]
     },                                                               
     :s => ["", "\n"]                                                 
   }                                                                  
 }}                                                                   
```

I don't know if I can provide a sample app, since I have no idea what is triggering this.

### Expected behavior

The tests pass like they did before.

### PR DESCRIPTION
**Pull request by derekkraan**
Fixes #1247

### CODE REVIEW

File: lib/phoenix_live_view/test/client_proxy.ex [hunk 1]
```diff
@@ -517,7 +517,8 @@ defmodule Phoenix.LiveViewTest.ClientProxy do
         payload = %{"cids" => will_destroy_cids}
 
         push_with_callback(state, view, "cids_will_destroy", payload, fn _, state ->
-          payload = %{"cids" => will_destroy_cids -- DOM.component_ids(view.id, state.html)}
+          still_there_cids = DOM.component_ids(view.id, state.html)
+          payload = %{"cids" => Enum.reject(will_destroy_cids, &(Enum.member?(still_there_cids, &1)))}
```
Comment (josevalim) [inline lib/phoenix_live_view/test/client_proxy.ex:521]:
```suggestion
          payload = %{"cids" => Enum.reject(will_destroy_cids, &(&1 in still_there_cids, &1))}
```

File: lib/phoenix_live_view/test/client_proxy.ex [hunk 2]
```diff
@@ -517,7 +517,8 @@ defmodule Phoenix.LiveViewTest.ClientProxy do
         payload = %{"cids" => will_destroy_cids}
 
         push_with_callback(state, view, "cids_will_destroy", payload, fn _, state ->
-          payload = %{"cids" => will_destroy_cids -- DOM.component_ids(view.id, state.html)}
+          still_there_cids = DOM.component_ids(view.id, state.html)
+          payload = %{"cids" => Enum.reject(will_destroy_cids, &(&1 in still_there_cids, &1))}
```
Comment (josevalim) [inline lib/phoenix_live_view/test/client_proxy.ex:521]:
```suggestion
          payload = %{"cids" => Enum.reject(will_destroy_cids, &(&1 in still_there_cids))}
```

My bad. :D

**Comment by derekkraan** [comment]
Haha I was about to go fix it but you beat me to it.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**